app/Main.hs

```diff
@@ -431,30 +431,30 @@ verify inputName = case mkPackageName (pack inputName) of
     pkg <- readPackageFile
     db <- readPackageSet pkg
     case name `Map.lookup` db of
-      Nothing -> echoT . pack $ "No packages found with the name " <> show (runPackageName $ name)
+      Nothing -> echoT . pack $ "No packages found with the name " <> show (runPackageName name)
       Just _ -> do
         reverseDeps <- map fst <$> getReverseDeps db name
         let packages = pure name <> reverseDeps
-        echoT ("Verifying " <> pack (show (length packages)) <> " packages.")
-        echoT "Warning: this could take some time!"
-
-        let installOrUpdate' (name_, pkgInfo) = (name_, ) <$> installOrUpdate (set pkg) name_ pkgInfo
-        paths <- Map.fromList <$> traverse installOrUpdate' (Map.toList db)
+        paths <- update (length packages) db pkg
 
         traverse_ (verifyPackage db paths) packages
 
 verifyPackageSet :: IO ()
 verifyPackageSet = do
   pkg <- readPackageFile
   db <- readPackageSet pkg
-  echoT ("Verifying " <> pack (show (Map.size db)) <> " packages.")
-  echoT "Warning: this could take some time!"
-
-  let installOrUpdate' (name, pkgInfo) = (name, ) <$> installOrUpdate (set pkg) name pkgInfo
-  paths <- Map.fromList <$> traverse installOrUpdate' (Map.toList db)
+  paths <- update (Map.size db) db pkg
 
   for_ (Map.toList db) $ \(name, _) -> verifyPackage db paths name
 
+update :: Int -> PackageSet -> PackageConfig -> IO (Map.Map PackageName Turtle.FilePath)
+update size db pkg = do
+  echoT ("Verifying " <> pack (show size) <> " packages.")
```

Comment:
We're not "verifying" here, this should say "updating" or "installing".

Reply:
I think "Verifying" is correct, the mistake was in naming the function `update`. I've moved things around a bit and renamed it.